source/directives/activecode.rst

```diff
@@ -0,0 +1,642 @@
+for unique identifiers in directives.
+
+Optional Arguments
+------------------
+content area
+    ``String``. The content area is a language dependent formatted string.
+
+    Content may be null and can be logically divided into 3 distinct regions,
+    all of which are optional.
+
+    The region above the ``~~~~`` delimiter defines a block of text that is not treated as source code.
+    It can contain any text as well as inline or paragraph level markup.
+    This block can be as large as you like and will be displayed at the top of the rendered output,
+    before the *Run* button.
+
+    The region below the ``====`` delimiter defines a block of text that is treated as source code,
+    but not displayed in the code editor.
+
+    All other content is treated as source code and displayed in the code editor.
+
+    .. code-block:: rst
+
+        .. activecode:: ac_ex_content
+           :language: python
+
+           Fix the following code so that it **always** correctly adds ``two`` numbers.
+
+           * Your solution must use the parameters a and b
+           ~~~~
+           def add(a,b):
+               return 4
+
+           ====
+           # TODO: add unit tests
+
+autograde
+    ``Boolean``. Declare an ``activecode`` block that will run unittest. Python only.
+
+autorun 
+    ``Boolean``. Declare an ``activecode`` block that will begin running as soon as 
+    the web page is fully loaded.
+
+available_files
+    ``CSV``. A comma separated list of simulated input files for use.
+
+    Each entry must be the unique_id of a :doc:`datafile` directive.
+
+caption
+    ``String``. Define a caption for the bottom of the directive frame.
+
+chatcodes
+    ``Boolean``. Enable users to talk about this code snippet with others
+
+compileargs
+    ``Array``. A list of compiler flags, passed to the external server with your code.
+    One string per array element.
+
+    This option is only used if the language uses a compiler: C and C++.
+    For example:
+
+    .. code-block:: rst
+
+       :language: cpp
+       :compileargs: ['-Wall', '-Wextra', '-pedantic', '-std=c++11']
+
+datafile
+    ``CSV``. A comma separated list of simulated input files for use.
+
+    Each entry must be the unique_id of a :doc:`datafile` directive.
+
+enabledownload
+    ``Boolean``. allow textfield contents to be downloaded to local file
+
+hidecode 
+    ``Boolean``. Make the ``activecode`` editor initially hidden, 
+    and add a button to automatically show the editor. 
+
+    When code is hidden, the 'Run' button is also disabled.
+
+    To run the code, either press 'Show Code', then 'Run',
+    or use the ``autorun`` option to run on page load.
+    
+    You might use this if you want to put an ``activecode`` block in the page in order to include 
+    it in another ``activecode`` block, but you don't need or want students to see it right away.
+
+include
+    ``CSV``. Include another ``activecode`` content area verbatim.
```

Comment:
What does CSV mean in this context??

I think an example showing how to specify an include would be useful and to note they are coma separated

Reply:
done.